engine/battle/ai/scoring.asm

```diff
@@ -1,10 +1,39 @@
 AIScoring: ; used only for BANK(AIScoring)
 
+; Details about diagrams :
+; 0) A flow is a succession of conditions and descriptions.
+; 1) A description is a sentence. It unconditionally refines a flow.
+;      A description can include odds of success.
+;      Examples : "for each other move", "50% to encourage".
+; 2) A condition is a question between parentheses. The condition must be met
+;      to switch to that flow.
+;      Example : "(is player immune)".
+; 3) When needed, a sentence can be split between lines with " ... ".
+;      Example : (is super effective...)
+;                (... or neutral)
+; 4) A branch, in " ┤ ┴ ┬ ┼ ├ ", represents an OR statement. It is followed by
+;      a condition at each end.
+; 5) The symbols " ► ◄ " is followed by the end of a flow.
+;      The latter is only used to save horizontal space.
+; 6) The symbol " & " represents multiple score changes for a single flow.
+;      Example : move ─&─► encourage
+;                      └ other moves ─► discourage
+;      reads as : move is encouraged and all others are discouraged.
+; 7) A jump to a different part of the graph is written between brackets.
+;      Example : (enemy is full HP) ─ [A] 
+;      reads as : " if enemy is full HP, jump to label [A] below.
 
 AI_Basic:
-; Don't do anything redundant:
-;  -Using status-only moves if the player can't be statused
-;  -Using moves that fail if they've already been used
+; Modifiable scores : ALL_MOVES | Randomness : NO
+
+; Don't do anything redundant :
+; - Using status-only moves if the player can't be statused.
+; - Using moves that fail if they've already been used.
+
+;  for each move ┬ (is redundant) ─► dismiss (-10)
+;                └ (not) ┬ (can't status) ─► dismiss (-10)
+;                        └ (can) ┬ (player has active Safeguard) ─► dismiss (-10)   
+;                                └ (not) ─► nothing
```

Comment:
Why is dismiss -10 here and +10 in other spots?

Reply:
Typo, thanks for spotting it.